sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs

```diff
@@ -145,8 +169,122 @@ internal static DetectLanguageResultCollection ConvertToDetectLanguageResultColl
         }
 
         #endregion
+
         #region AnalyzeSentiment
 
+        internal static TextSentiment ConvertToTextSentiment(Legacy.Models.SentenceSentimentValue sentiment) =>
+            sentiment switch
+            {
+                Legacy.Models.SentenceSentimentValue.Neutral => TextSentiment.Neutral,
+                Legacy.Models.SentenceSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.SentenceSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The sentence sentiment, { sentiment } is not supported for conversion.")
+            };
+
+        internal static TextSentiment ConvertToTextSentiment(Legacy.Models.TokenSentimentValue sentiment) =>
+            sentiment switch
+            {
+                Legacy.Models.TokenSentimentValue.Mixed => TextSentiment.Neutral,
+                Legacy.Models.TokenSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.TokenSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The token sentiment, { sentiment } is not supported for conversion.")
+            };
+
+        internal static TextSentiment ConvertToTextSentiment(Legacy.Models.DocumentSentimentValue sentiment) =>
+            sentiment switch
+            {
+                Legacy.Models.DocumentSentimentValue.Neutral => TextSentiment.Neutral,
+                Legacy.Models.DocumentSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.DocumentSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The document sentiment, { sentiment } is not supported for conversion.")
+            };
+
+        //TODO:  Check this conversion.  Seems to match: https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment.cs#L56
```

Comment:
@maririos, @kinelski:   I could use your expertise here.  From what I can tell, this is the same conceptual transform that was being performed.   But, the existing transformation is odd and I'm not sure that I'm interpreting it correctly.